cluster/images/conformance/Makefile

```diff
@@ -18,20 +18,52 @@
 #   [ARCH=amd64] [REGISTRY="k8s.gcr.io"] make (build|push) VERSION={some_released_version_of_kubernetes}
 
 REGISTRY?=k8s.gcr.io
+OS?=linux
+OS_VERSION?=1809
 ARCH?=amd64
 OUT_DIR?=_output
 
-LOCAL_OUTPUT_PATH=$(shell pwd)/../../../$(OUT_DIR)/local/bin/linux/$(ARCH)
-DOCKERIZED_OUTPUT_PATH=$(shell pwd)/../../../$(OUT_DIR)/dockerized/bin/linux/$(ARCH)
+# The output type could either be docker (local), or registry.
+# If it is registry, it will also allow us to push the Windows images.
+OUTPUT_TYPE = docker
+ifeq ($(MAKECMDGOALS),push)
+  OUTPUT_TYPE = registry
+else ifeq ($(OS),windows)
+    $(error Windows images can only be created and pushed directly. Use make push for Windows images)
+endif
+
+IMAGE_NAME_EXTENSION.linux = $(OS)-$(ARCH)
+IMAGE_NAME_EXTENSION.windows = $(OS)-$(ARCH)-$(OS_VERSION)
+IMAGE_NAME_EXTENSION := ${IMAGE_NAME_EXTENSION.${OS}}
+
+EXTENSION.linux =
+EXTENSION.windows = .exe
+EXTENSION := ${EXTENSION.${OS}}
+
+GINKGO := ginkgo$(EXTENSION)
+KUBECTL := kubectl$(EXTENSION)
+E2E_TEST := e2e.test$(EXTENSION)
+GO_RUNNER := go-runner$(EXTENSION)
+GORUNNER := gorunner$(EXTENSION)
 
-GINKGO_BIN?=$(shell test -f $(LOCAL_OUTPUT_PATH)/ginkgo && echo $(LOCAL_OUTPUT_PATH)/ginkgo || echo $(DOCKERIZED_OUTPUT_PATH)/ginkgo)
-KUBECTL_BIN?=$(shell test -f $(LOCAL_OUTPUT_PATH)/kubectl && echo $(LOCAL_OUTPUT_PATH)/kubectl || echo $(DOCKERIZED_OUTPUT_PATH)/kubectl)
-E2E_TEST_BIN?=$(shell test -f $(LOCAL_OUTPUT_PATH)/e2e.test && echo $(LOCAL_OUTPUT_PATH)/e2e.test || echo $(DOCKERIZED_OUTPUT_PATH)/e2e.test)
-E2E_GO_RUNNER_BIN?=$(shell test -f $(LOCAL_OUTPUT_PATH)/go-runner && echo $(LOCAL_OUTPUT_PATH)/go-runner || echo $(DOCKERIZED_OUTPUT_PATH)/go-runner)
+LOCAL_OUTPUT_PATH=$(shell pwd)/../../../$(OUT_DIR)/local/bin/$(OS)/$(ARCH)
+DOCKERIZED_OUTPUT_PATH=$(shell pwd)/../../../$(OUT_DIR)/dockerized/bin/$(OS)/$(ARCH)
+
+GINKGO_BIN?=$(shell test -f $(LOCAL_OUTPUT_PATH)/$(GINKGO) && echo $(LOCAL_OUTPUT_PATH)/$(GINKGO) || echo $(DOCKERIZED_OUTPUT_PATH)/$(GINKGO))
+KUBECTL_BIN?=$(shell test -f $(LOCAL_OUTPUT_PATH)/$(KUBECTL) && echo $(LOCAL_OUTPUT_PATH)/$(KUBECTL) || echo $(DOCKERIZED_OUTPUT_PATH)/$(KUBECTL))
+E2E_TEST_BIN?=$(shell test -f $(LOCAL_OUTPUT_PATH)/$(E2E_TEST) && echo $(LOCAL_OUTPUT_PATH)/$(E2E_TEST) || echo $(DOCKERIZED_OUTPUT_PATH)/$(E2E_TEST))
+E2E_GO_RUNNER_BIN?=$(shell test -f $(LOCAL_OUTPUT_PATH)/$(GO_RUNNER) && echo $(LOCAL_OUTPUT_PATH)/$(GO_RUNNER) || echo $(DOCKERIZED_OUTPUT_PATH)/$(GO_RUNNER))
 
 CLUSTER_DIR?=$(shell pwd)/../../../cluster/
 
-BASEIMAGE=debian:stable-slim
+BASEIMAGE.linux=debian:stable-slim
+BASEIMAGE.windows=mcr.microsoft.com/windows/nanoserver:$(OS_VERSION)
```

Comment:
So I think you're going to run into the same problem I did with nanoserver; a necessary dll is missing. See https://github.com/docker-library/golang/issues/348 The first comment on there is right though: it can build but will panic at runtime.

The fix, as they said, is to copy the dll from servercore. Otherwise if you just change to servercore the image is 10x the size.

Just did this for sonobuoy: https://github.com/vmware-tanzu/sonobuoy/blob/master/DockerfileWindows#L18